homeassistant/components/recorder/statistics.py

```diff
@@ -2069,21 +2079,33 @@ def _sorted_statistics_to_dict(
         ent_results = result[statistic_id]
         for db_state in stats_list:
             row: dict[str, Any] = {
-                "start": (start_ts := db_state.start_ts),
+                "start": (start_ts := db_state[start_ts_idx]),
                 "end": start_ts + table_duration_seconds,
             }
-            if _want_mean:
-                row["mean"] = convert(db_state.mean) if convert else db_state.mean
-            if _want_min:
-                row["min"] = convert(db_state.min) if convert else db_state.min
-            if _want_max:
-                row["max"] = convert(db_state.max) if convert else db_state.max
             if _want_last_reset:
-                row["last_reset"] = db_state.last_reset_ts
-            if _want_state:
-                row["state"] = convert(db_state.state) if convert else db_state.state
-            if _want_sum:
-                row["sum"] = convert(db_state.sum) if convert else db_state.sum
+                row["last_reset"] = db_state[last_reset_ts_idx]
+            if convert:
+                if _want_mean:
+                    row["mean"] = convert(db_state[mean_idx])
+                if _want_min:
+                    row["min"] = convert(db_state[min_idx])
+                if _want_max:
+                    row["max"] = convert(db_state[max_idx])
+                if _want_state:
+                    row["state"] = convert(db_state[state_idx])
+                if _want_sum:
+                    row["sum"] = convert(db_state[sum_idx])
+            else:
+                if _want_mean:
+                    row["mean"] = db_state[mean_idx]
+                if _want_min:
+                    row["min"] = db_state[min_idx]
+                if _want_max:
+                    row["max"] = db_state[max_idx]
+                if _want_state:
+                    row["state"] = db_state[state_idx]
+                if _want_sum:
+                    row["sum"] = db_state[sum_idx]
```

Comment:
Its longer and has more branching but I think its easier to read and is faster when we ask for multiple since we only check convert once per loop